### PR TITLE
feat: 3id, env vars, dids use local ipfs

### PR DESCRIPTION
**Pull request by zachferland**
**Changes:**
Add 3id
Resolve both 3id and muport through our ipfs instance 
allow env vars to overwrite kms 

**Additional deployment configurations:**
- up lambda execution role to access s3 resources (before)
- alter pg table (before )
ALTER TABLE public.links ALTER COLUMN did TYPE VARCHAR(100);

**Tests:**
There was more tests than i thought, but none to actually test resolvers here ( s3 stuff or ipfs mock). Probably not worth adding at moment, may add some tests for the basic config changes. API service tests most of that other code, and once shared part pulled out, then exact code will have same coverage. 

**TODO:**
release and reference ipfs-s3-dag-get instead local file, remove dependencies included there instead.

### CODE REVIEW

File: src/lib/uPortMgr.js [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
+import { initIPFS } from "./ipfs-fs-to-s3-datastore.js"
+const register3idResolver = require('3id-resolver')
+const registerMuPortResolver = require("muport-did-resolver")
+const registerUPortResolver = require("uport-did-resolver")
 
-require("uport-did-resolver")();
-require("muport-did-resolver")();
+// Register resolvers
+function register (ipfs) {
+  registerUPortResolver()
```
Comment (oed) [inline src/lib/uPortMgr.js]: This can be removed.

File: src/api_handler.js [hunk 2]
```diff
@@ -60,17 +60,27 @@ const doHandler = (handler, event, context, callback) => {
...
+  if (!addressMgr.isSecretsSet() || !linkMgr.isSecretsSet() || uPortMgr.isSecretsSet()) {
     const kms = new AWS.KMS()
     kms
       .decrypt({ CiphertextBlob: Buffer(process.env.SECRETS, 'base64') })
       .promise()
       .then(data => {
         const decrypted = String(data.Plaintext)
-        addressMgr.setSecrets(JSON.parse(decrypted))
-        linkMgr.setSecrets(JSON.parse(decrypted))
+        const config = Object.assign(JSON.parse(decrypted), envConfig)
```
Comment (oed) [inline src/api_handler.js]: Why not only use the process.env variables?
